Library/Homebrew/cmd/uses.rb

```diff
@@ -47,48 +47,49 @@ def uses
 
     uses = formulae.select do |f|
       used_formulae.all? do |ff|
-        begin
-          if recursive
-            deps = f.recursive_dependencies do |dependent, dep|
-              if dep.recommended?
-                Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
-              elsif dep.optional?
-                Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
-              elsif dep.build?
-                Dependency.prune unless includes.include?("build?")
-              end
+        if recursive
+          deps = f.recursive_dependencies do |dependent, dep|
+            if dep.recommended?
+              Dependency.prune if ignores.include?("recommended?") || dependent.build.without?(dep)
+            elsif dep.optional?
+              Dependency.prune if !includes.include?("optional?") && !dependent.build.with?(dep)
+            elsif dep.build?
+              Dependency.prune unless includes.include?("build?")
             end
-            reqs = f.recursive_requirements do |dependent, req|
-              if req.recommended?
-                Requirement.prune if ignores.include?("recommended?") || dependent.build.without?(req)
-              elsif req.optional?
-                Requirement.prune if !includes.include?("optional?") && !dependent.build.with?(req)
-              elsif req.build?
-                Requirement.prune unless includes.include?("build?")
-              end
-            end
-          else
-            deps = f.deps.reject do |dep|
-              ignores.any? { |ignore| dep.send(ignore) } && !includes.any? { |include| dep.send(include) }
-            end
-            reqs = f.requirements.reject do |req|
-              ignores.any? { |ignore| req.send(ignore) } && !includes.any? { |include| req.send(include) }
+
+            # If a tap isn't installed, we can't find the dependencies of one
+            # its formulae, and an exception will be thrown if we try.
+            if dep.is_a?(TapDependency) && !dep.tap.installed?
+              Dependency.keep_but_prune_recursive_deps
             end
```

Comment:
Because I removed an enclosing block and outdented its contents, the diff is a bit messy, so I'll highlight my actual code changes here.

This is where I added the check to avoid the exception when a dependency is untapped.